src/hotspot/cpu/x86/x86.ad

```diff
@@ -8973,6 +8978,77 @@ instruct vcompress_mask_reg_evex(kReg dst, kReg mask, rRegL rtmp1, rRegL rtmp2,
 
 #endif // _LP64
 
+// -------------------------------- Bit and Byte Reversal Vector Operations ------------------------
+instruct vreverse_reg_avx(vec dst, vec src, vec xtmp1, vec xtmp2, vec xtmp3, rRegI rtmp) %{
+  predicate(!VM_Version::supports_avx512bw() && Matcher::vector_length_in_bytes(n) < 64  && !VM_Version::supports_gfni());
+  match(Set dst (ReverseV src));
+  effect(TEMP dst, TEMP xtmp1, TEMP xtmp2, TEMP xtmp3, TEMP rtmp);
+  format %{ "vector_reverse_bit_avx $dst, $src!\t using $xtmp1, $xtmp2, $xtmp3 and $rtmp as TEMP" %}
+  ins_encode %{
+    int vec_enc = vector_length_encoding(this);
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    __ vector_reverse_bit(bt, $dst$$XMMRegister, $src$$XMMRegister, $xtmp1$$XMMRegister,
+                          $xtmp2$$XMMRegister, $xtmp3$$XMMRegister, $rtmp$$Register, vec_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vreverse_reg_evex(vec dst, vec src, vec xtmp1, vec xtmp2, rRegI rtmp) %{
+  predicate((VM_Version::supports_avx512bw() || Matcher::vector_length_in_bytes(n) == 64) && !VM_Version::supports_gfni());
```

Comment:
If vector length < 64 we need avx512vl here. I think this and the above should be `VM_Version::supports_avx512vlbw()` instead

Reply:
Even though there is no X86 target which supports BW but not VL,  but we still need to be granular in emitting correct predications since one can always configure VMs to enable selective target features.